flask_cors/__init__.py

```diff
@@ -238,17 +244,25 @@ def cors_after_request(resp):
                 The actual after-request handler, retains references to the
                 the options, and definitions of resources through a closure.
             '''
+            logger.debug('Checking CORS for request')
             # If CORS headers are set in a view decorator, pass
             if resp.headers.get(ACL_ORIGIN):
+                logger.debug('Header {} have been set already, skipping'.format(ACL_ORIGIN))
                 return resp
 
+            logger.debug('Checking available CORS rules')

```

Comment:
I also suggest removing this line, as it will be written out on every request :-(